src/core/importType.js

```diff
@@ -109,20 +109,23 @@ function isExternalLookingName(name) {
   return isModule(name) || isScoped(name);
 }
 
-function typeTest(name, context, path ) {
+function typeTest(name, context, getPath) {
   const { settings } = context;
   if (isInternalRegexMatch(name, settings)) { return 'internal'; }
-  if (isAbsolute(name, settings, path)) { return 'absolute'; }
+  if (isAbsolute(name)) { return 'absolute'; }
+  if (isRelativeToParent(name)) { return 'parent'; }
+  if (isIndex(name)) { return 'index'; }
+  if (isRelativeToSibling(name)) { return 'sibling'; }
+
+  // `resolve` is expensive, do it only when necessary
+  const path = getPath();
   if (isBuiltIn(name, settings, path)) { return 'builtin'; }
```

Comment:
this change moves `isBuiltIn` below absolute/relativeToParent/index/relativeToSibling - this might be a breaking change for anyone relying on the `import/core-modules` setting, which currently takes precedence.

Reply:
Could we make this change in next major version?

Reply:
There's unlikely to be a major version any time soon, if ever.